src/browser/api/window.js

```diff
@@ -1948,13 +1949,30 @@ function saveBoundsToDisk(identity, bounds, zoomLevel, callback) {
     } catch (err) {
         callback(err);
     }
-
 }
+
 //make sure the uuid/names with special characters do not break the bounds cache.
 function getBoundsCacheSafeFileName(identity) {
-    let safeName = new Buffer(identity.uuid + '-' + identity.name).toString('hex');
     const userCache = electronApp.getPath('userCache');
-    return path.join(userCache, windowPosCacheFolder, `${safeName}.json`);
+
+    // new style file name
+    const hash = crypto.createHash('sha256');
+    hash.update(identity.uuid);
+    hash.update(identity.name);
+    const safeName = hash.digest('hex');
+    const newFileName = path.join(userCache, windowPosCacheFolder, `${safeName}.json`);
+
+    if (!fs.existsSync(newFileName)) {
```

Comment:
Just a thought, if we know that this does not exist we could let the caller know, as often it will immediately try to read it (see [here](https://github.com/HadoukenIO/core/blob/46e136b0a9c052461e2fa4859b29e9a1b254eb41/src/browser/api/window.js#L1766)). Though I think this is fine as is.

Reply:
yeah, i thought that too. But i found this  function 'getBoundsCacheSafeFileName" is also called in saveBoundsToDisk  to get the cached file name.  If i want to return something not "cached file name" when it doesn't exist, i need to add something else to differentiate  these two cases.